docs/polkadot/implementors-guide/guide.md

```diff
@@ -0,0 +1,342 @@
+1. A relay-chain block author, selected by BABE, can include up to one (1) backed candidate for each parachain to include in the relay-chain block alongside its backing.
+1. Once included, the parachain candidate is considered to be "pending availability". It is not considered fully included until it is proven available.
+1. In the following relay-chain blocks, validators will participate in the Availability Distribution Process to ensure availability of the candidate. Information regarding the availability of the candidate will be included in the subsequent relay-chain blocks.
+1. Once the relay-chain state machine has enough information to consider the candidate's PoV as being available, the candidate is considered fully included and is graduated to being a full parachain block, or parablock for short.
+
+Note that the candidate can fail to be included in any of the following ways:
+  - The collator is not able to propagate the candidate to any validators.
+  - The candidate is not fully backed by validators participating in the Candidate Backing Process.
+  - The candidate is not selected by a relay-chain block author to be included in the relay chain
+  - The candidate's PoV is not considered as available within a timeout and is discarded from the relay chain.
+
+This process can be divided further down. Steps 2 & 3 relate to the work of the collator in collating and distributing the candidate to validators via the Collation Distribution Process. Steps 3 & 4 relate to the work of the validators in the Candidate Backing Process and the block author (itself a validator) to include the block into the relay chain. Steps 6, 7, and 8 correspond to the logic of the relay-chain state-machine (otherwise known as the Runtime) used to fully incorporate the block into the chain. Step 7 requires further work on the validators' parts to participate in the Availability Distribution process and include that information into the relay chain for step 8 to be fully realized.
+
+This brings us to the second part of the process. Once a parablock is "fully included", it is still "pending approval". At this stage in the pipeline, the parablock has been backed by a majority of validators in the group assigned to that parachain, and its data has been guaranteed available by the set of validators as a whole. However, the validators in the parachain-group (known as the "Parachain Validators" for that parachain) are sampled from a validator set which contains some proportion of byzantine, or arbitrarily malicious members. This implies that the Parachain Validators for some parachain may be majority-dishonest, which means that secondary checks must be done on the block before it can be considered approved.
+
+The Approval Process looks like this:
```

Comment:
Yeah, long story short is that you, over time, need to assign  >33% of your validators to check a specific parachain block. You can do that all at once in the first phase, but then you aren't sharding or scaling. The second phase (where we do secondary checks _after_ optimistic inclusion) gives us optimistic scaling with identical security properties.

I'll let @burdges, @FatemeShirazi, or @AlistairStewart add to that with any more details or correct me if I'm wrong.

This is definitely important background information, so I'll look into adding a short-and-sweet introduction to the sharding approach. Although, I don't think that we could put all the necessary background information into this doc - it would just be too much.

Reply:
(I added the sharding intro - do you consider this resolved?)

Reply:
Even with random assignment in order to have only one pipeline, the number of parachain validators assigned to a parachain needs to be quite large to make sure for every parachain majority of assigned parachain validators is honest. As Rob mentioned, having two pipelines allows us to get away with scaling very well in case validators are honest and having an assurance that if that is not the case it will be caught by the second pipeline. The second pipeline has to happen anyway also, once we have any fishermen reporting faulty behavior. In fact, the more reports we have from fishermen the number of secondary checkers increases.

Reply:
>long story short is that you, over time, need to assign >33% of your validators to check a specific parachain block.

This is not true as written. Without challenges, only a few validators check a specific parachain block.

The goal here is to make attacking Polkadot expensive, while scaling as much as possible. We can get stronger security with fewer checkers with this split into backing and approval checkers as Fatemeh explained.